src/network/Network.ts

```diff
@@ -5,34 +5,66 @@ export default class Network {
 
     private falconEye: FalconEye;
     private address: string;
-    private profileId: string;
 
     constructor(falconEye: FalconEye, address: string) {
         this.falconEye = falconEye;
-        this.address = address;
-        this.profileId = falconEye.getConfig().profileId;
+        this.address = address + (address.slice(-1) === '/' ? "" : "/");
     }
 
-    public async handshake(): Promise<void> {
-        let This = this;
-
-        await request({
+    public async handshake(): Promise<boolean> {
+        return request({
             method: 'POST',
-            uri: this.address + (this.address.slice(-1) === '/' ? "" : "/") +  "connect/",
+            uri: this.address + "connect/",
             body: {
-                apikey: this.falconEye.getAPIKey()
+                apiKey: this.falconEye.getConfig().apiKey
             },
             json: true
         })
         .then((parsedBody: any) => {
-            This.profileId = parsedBody.profileid;
+            let code = parsedBody.code;
+
+            if (code === "00") {
```

Comment:
Since Error codes from the server are unique, a separate class could handle them